array_methods_assignmen.js

```diff
@@ -0,0 +1,404 @@
+// # Sihame Bazi Array methods lab
+//
+// 1. Given an array of numbers, write a function that
+//returns a new array where every element is
+//incremented by 10.
+//
+// ```
+
+// Input: [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+//
+// let array = [1,2,3,4,5];
+// let newArray = array.map((currentElement) =>{
+//    return currentElement + 10
+//  })
+//  console.log(newArray)
+// ============================================
+// 2. Given an array, return a new array where
+//every non-string element is converted
+//to an empty string.
+//
+// ```
+// Input: ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+//
+// let arr2 = ["a", 123, "b", "c", {name: "cat"}];
+// let allStringArr = arr2.map((crrElem, index) => {
+// if (typeof crrElem === 'number' || typeof crrElem === 'object' ) {
+//  return " "
+// } else {
+//  return  crrElem
+// }
+// })
+// console.log(allStringArr)
+
```

Comment:
What if the typeof was a boolean? What way can we make sure we can attest that if it is not a string.